### PR TITLE
List resources a user has access to in the sidebar.

### PR DESCRIPTION
**Pull request by n1k0**
Paired with @magopian, refs https://github.com/Kinto/kinto/pull/653

I didn't deploy to gh-pages because it'd need the feature to be installed on the dev server.

/cc @leplatrem @Natim @glasserc

### CODE REVIEW

File: scripts/client.js [hunk 1]
```diff
@@ -80,3 +80,28 @@ export function requestAttachment(
...
+
+export function requestPermissions(): Promise {
+  const client: KintoClient = getClient();
+  if (!client) {
+    throw new Error("Client is not configured.");
+  }
+  const {remote, defaultReqOptions}: {
+    remote: string,
+    defaultReqOptions: {
+      headers: Object
```
Comment (Natim) [inline scripts/client.js]: Is Object better than {} here?

File: scripts/client.js [hunk 2]
```diff
@@ -80,3 +80,28 @@ export function requestAttachment(
...
+    }
+  } = client;
+  return fetch(`${remote}/permissions`, {
```
Comment (Natim) [inline scripts/client.js]: Shouldn't we add that to kinto-http.js ?

**Comment by Natim** [comment]
f+

File: scripts/client.js (hunk 1 above)
Comment (n1k0) [inline scripts/client.js]: This is a flow type.

File: scripts/client.js (hunk 2 above)
Comment (n1k0) [inline scripts/client.js]: Yes we should, just not now.

**Comment by n1k0** [comment]
We'll badly need to add functional tests there.

File: scripts/sagas/session.js [hunk 3]
```diff
@@ -47,12 +47,34 @@ export function* listBuckets(getState, action) {
     const responses = yield call([client, client.batch], (batch) => {
       for (const {id} of data) {
         batch.bucket(id).listCollections();
+        // XXX future: query the /permissions endpoint to retrieve the other
+        // collections we may have access to bcause they're shared with us
```
Comment (leplatrem) [inline scripts/sagas/session.js]: typo because

File: scripts/sagas/session.js [hunk 4]
```diff
@@ -47,12 +47,34 @@ export function* listBuckets(getState, action) {
...
+    // the /permissions endpoint
+    for (const permission of permissions.data) {
+      // Add any missing bucket to the current list
+      let bucket = buckets.find(x => x.id === permission.bucket_id);
+      if (!bucket) {
+        bucket = {id: permission.bucket_id, collections: []};
+        buckets.push(bucket);
+      }
+      // Add any missing collection to the current bucket collections list
+      if ("collection_id" in permission) {
```
Comment (leplatrem) [inline scripts/sagas/session.js]: Add a comment that it covers records and collections

**Comment by leplatrem** [comment]
Nicely done!

This will also work for individual records that are shared in private collections :) Also because the server will return a partial records list when querying a collection where only a couple of records are shared :)

One of the next steps that comes into my mind (further issue/pr) would be to show edit button on buckets/collection/records only if write is granted! Awesome :)

File: scripts/sagas/session.js [hunk 5]
```diff
@@ -47,12 +47,34 @@ export function* listBuckets(getState, action) {
...
         batch.bucket(id).listCollections();
+        // XXX future: query the /permissions endpoint to retrieve the other
+        // collections we may have access to bcause they're shared with us
       }
     });
     const buckets = data.map((bucket, index) => {
       const collections = responses[index].body.data;
       return {id: bucket.id, collections};
     });
+    const permissions = yield call(requestPermissions);
```
Comment (leplatrem) [inline scripts/sagas/session.js]: Having a `if` here for a (capability/protocol version) check wouldn't be too much hassle actually

Comment (n1k0) [inline scripts/sagas/session.js]: Well, indeed, though we should probably have a capability exposed in the first place :)

Comment (leplatrem) [inline scripts/sagas/session.js]: If we make it into the core, the check will have to be made on `http_protocol` version number

**Comment by n1k0** [comment]
> One of the next steps that comes into my mind (further issue/pr) would be to show edit button on buckets/collection/records only if write is granted!

Please note that even if we display these, if you don't have write permissions you'll get a warning rendered and disabled edit forms.

File: scripts/sagas/session.js (hunk 5 above)
Comment (n1k0) [inline scripts/sagas/session.js]: Ah, right.

**Comment by n1k0** [comment]
@leplatrem r=? for https://github.com/Kinto/kinto-admin/pull/161/commits/5a167983e98334c0d485933b1a464418737d0729

**Comment by n1k0** [comment]
This has been pair-reviewed with @magopian and @leplatrem.

**Comment by leplatrem** [comment]
r+!

**Pull request merged**